### DISCUSSION TITLE
Writinng a plugin for fetching data from an API

### DISCUSSION THREAD

**Discussion by Temidayo32** [getpelican/pelican#3060] [Q&A]
Hi, I need to write a plugin that allows me fetch data from a GRAPHQl API (A headless CMS actually). I need Pelican as a presentation layer. This link seems to be the only pointer to that, but the illustration does not provide much help.
https://docs.getpelican.com/en/latest/plugins.html#using-plugins-to-inject-content

**Comment by galaxy4public** [comment]
Here is a copy of my internal "social" plugin I was using to update twitter counters - https://gist.github.com/galaxy4public/b817e8191c9c05bee0062dc8aef93e9a - I hope you will find it useful for you purposes, since approach is exactly the same: call an API and integrate the results into Pelican.
